src/azul/service/source_controller.py

```diff
@@ -32,6 +36,10 @@ class SourceController(Controller):
     def _source_service(self) -> SourceService:
         return SourceService()
 
+    @property
+    def service(self) -> ElasticsearchService:
+        raise NotImplementedError
```

Comment:
Smells. Try pushing _add_implicit_sources_filter down instead of pulling service up.

https://refactoring.com/catalog/pushDownMethod.html
https://refactoring.com/catalog/pullUpMethod.html

Reply:
That will cause problems once we apply the managed access controls to the manifest endpoints (https://github.com/DataBiosphere/azul/pull/3309), because  `ManifestController` will need access to `_add_implicit_sources_filter` as well as `RepositoryController`.